common/sgx/eeid.c

```diff
@@ -0,0 +1,499 @@
+        &remaining,
+        (uint8_t*)eeid->hash_state.N,
+        sizeof(eeid->hash_state.N)));
+    OE_CHECK(serialize_element(
+        "signature_size",
+        position,
+        &remaining,
+        (uint8_t*)&eeid->signature_size,
+        sizeof(eeid->signature_size)));
+    OE_CHECK(serialize_element(
+        "signature",
+        position,
+        &remaining,
+        eeid->signature,
+        eeid->signature_size));
+    OE_CHECK(serialize_element(
+        "size_settings",
+        position,
+        &remaining,
+        (uint8_t*)&eeid->size_settings,
+        sizeof(eeid->size_settings)));
+    OE_CHECK(serialize_element(
+        "data_size",
+        position,
+        &remaining,
+        (uint8_t*)&eeid->data_size,
+        sizeof(eeid->data_size)));
+    OE_CHECK(serialize_element(
+        "vaddr",
+        position,
+        &remaining,
+        (uint8_t*)&eeid->vaddr,
+        sizeof(eeid->vaddr)));
+    OE_CHECK(serialize_element(
+        "entry_point",
+        position,
+        &remaining,
+        (uint8_t*)&eeid->entry_point,
+        sizeof(eeid->entry_point)));
+    OE_CHECK(serialize_element(
+        "data", position, &remaining, (uint8_t*)eeid->data, eeid->data_size));
+
+    **position = '\0';
+
+done:
+    return OE_OK;
+}
+
+// Note: this needs to be updated when oe_eeid_t changes.
+oe_result_t oe_deserialize_eeid(
+    const char* buf,
+    size_t buf_size,
+    oe_eeid_t* eeid)
+{
+    oe_result_t result;
+
+    if (!buf || !buf_size || !eeid)
+        return OE_INVALID_PARAMETER;
+
+    const char** position = &buf;
+    size_t remaining = buf_size;
+
+    memset(eeid, 0, sizeof(oe_eeid_t));
+
+    OE_CHECK(deserialize_element(
+        position,
+        &remaining,
+        (uint8_t*)eeid->hash_state.H,
+        sizeof(eeid->hash_state.H)));
+    OE_CHECK(deserialize_element(
+        position,
+        &remaining,
+        (uint8_t*)eeid->hash_state.N,
+        sizeof(eeid->hash_state.N)));
+    OE_CHECK(deserialize_element(
+        position,
+        &remaining,
+        (uint8_t*)&eeid->signature_size,
+        sizeof(eeid->signature_size)));
+    OE_CHECK(deserialize_element(
+        position, &remaining, eeid->signature, eeid->signature_size));
+    OE_CHECK(deserialize_element(
+        position,
+        &remaining,
+        (uint8_t*)&eeid->size_settings,
+        sizeof(eeid->size_settings)));
+    OE_CHECK(deserialize_element(
+        position,
+        &remaining,
+        (uint8_t*)&eeid->data_size,
+        sizeof(eeid->data_size)));
+    OE_CHECK(deserialize_element(
+        position, &remaining, (uint8_t*)&eeid->vaddr, sizeof(eeid->vaddr)));
+    OE_CHECK(deserialize_element(
+        position,
+        &remaining,
+        (uint8_t*)&eeid->entry_point,
+        sizeof(eeid->entry_point)));
+    OE_CHECK(deserialize_element(
+        position, &remaining, (uint8_t*)eeid->data, eeid->data_size));
+
+done:
+    return OE_OK;
+}
+
+oe_result_t oe_remeasure_memory_pages(
```

Comment:
This code is host-only right? If so, it shouldn't be in common

Reply:
No, it's truly common. E.g. the tests verify quotes in enclaves and on the host, and EEID requires re-measurement of the added memory pages, so this becomes common now.